group-coordinator/src/main/java/org/apache/kafka/coordinator/group/metrics/CoordinatorRuntimeMetrics.java

```diff
@@ -0,0 +1,244 @@
+     */
+    protected final MetricName numPartitionsInitial = getMetricName("NumPartitionsInitial");
+    /**
+     * Visible for testing.
+     */
+    public static final AtomicLong NUM_PARTITIONS_INITIAL = new AtomicLong(0);
+
+    /**
+     * Metric to count the number of partitions in Loading state.
+     */
+    protected final MetricName numPartitionsLoading = getMetricName("NumPartitionsLoading");
+    /**
+     * Visible for testing.
+     */
+    public static final AtomicLong NUM_PARTITIONS_LOADING = new AtomicLong(0);
+
+    /**
+     * Metric to count the number of partitions in Active state.
+     */
+    protected final MetricName numPartitionsActive = getMetricName("NumPartitionsActive");
+    /**
+     * Visible for testing.
+     */
+    public static final AtomicLong NUM_PARTITIONS_ACTIVE = new AtomicLong(0);
+
+    /**
+     * Metric to count the number of partitions in Closed state.
+     */
+    protected final MetricName numPartitionsClosed = getMetricName("NumPartitionsClosed");
+    /**
+     * Visible for testing.
+     */
+    public static final AtomicLong NUM_PARTITIONS_CLOSED = new AtomicLong(0);
+
+    /**
+     * Metric to count the number of partitions in Failed state.
+     */
+    protected final MetricName numPartitionsFailed = getMetricName("NumPartitionsFailed");
+    /**
+     * Visible for testing.
+     */
+    public static final AtomicLong NUM_PARTITIONS_FAILED = new AtomicLong(0);
+
+    protected MetricsRegistry registry;
+    protected Metrics metrics;
+    public Sensor partitionLoadSensor;
+
+    /**
+     * Add metrics to the registries.
+     */
+    protected void initializeMetrics() {
+        registry.newGauge(numPartitionsInitial, new Gauge<Long>() {
+            @Override
+            public Long value() {
+                return NUM_PARTITIONS_INITIAL.get();
+            }
+        });
+
+        registry.newGauge(numPartitionsLoading, new Gauge<Long>() {
+            @Override
+            public Long value() {
+                return NUM_PARTITIONS_LOADING.get();
+            }
+        });
+
+        registry.newGauge(numPartitionsActive, new Gauge<Long>() {
+            @Override
+            public Long value() {
+                return NUM_PARTITIONS_ACTIVE.get();
+            }
+        });
+
+        registry.newGauge(numPartitionsClosed, new Gauge<Long>() {
+            @Override
+            public Long value() {
+                return NUM_PARTITIONS_CLOSED.get();
+            }
+        });
+
+        registry.newGauge(numPartitionsFailed, new Gauge<Long>() {
+            @Override
+            public Long value() {
+                return NUM_PARTITIONS_FAILED.get();
+            }
+        });
+
+        this.partitionLoadSensor.add(
+            metrics.metricName(
+                "partition-load-time-max",
+                metricsGroup(),
+                "The max time it took to load the partitions in the last 30sec"
```

Comment:
Just curious about this, where do we set the 30s? Or it is just the default value?

Reply:
correct, it's the default value defined here: https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/metrics/MetricConfig.java#L39